core/output/slack.js

```diff
@@ -200,49 +247,45 @@ startPingPongTimer = function(teamId) {
 		messageId = ++team.ping_id;
 
 	if (!messageId) {
-		messageId = 0;
+		messageId = 1;
 		team.ping_id = messageId;
 	}
 
 	team.timeSincelastMessageRecieved = Date.now();
-	setTimeout(() => {
-		if (Date.now() - team.timeSincelastMessageRecieved >= 60000 && messageId === team.ping_id) {
+	let timeout = setTimeout(() => {
+		if (Date.now() - team.timeSincelastMessageRecieved >= 60000 && messageId === team.ping_id && !shuttingDown) {
 			console.debug('slack-> sending ping');
 			sendPing(teamId);
 		}
 	}, 60000);
+	timeouts.push(timeout);
 },
 
-sendPing = function(teamId) {
-	var socket = sockets[teamId],
-		slackTeams = exports.config.slack_teams,
+sendPing = teamId => {
+	let slackTeams = exports.config.slack_teams,
 		team = slackTeams[teamId],
 		pongId = ++team.pong_id;
 
 	if (!pongId) {
-		pongId = 0;
+		pongId = 1;
 		team.pong_id = pongId;
 	}
 
-	if (socket != null) {
-		var body = {
-			'id': team.pong_id,
-			'type':'ping',
-			'timestamp': Date.now()
-		};
-		body = JSON.stringify(body);
-		socket.send(body);
+	let body = {
+	'id': team.pong_id,
+	'type':'ping',
+	'timestamp': Date.now()
+	};
+	body = JSON.stringify(body);
+	sendSocketMessage(teamId, body, undefined);
```

Comment:
just use `sendSocketMessage(teamId, body)` not `sendSocketMessage(teamId, body, undefined)`.